viewing_party/party.py

```diff
@@ -1,23 +1,112 @@
-# ------------- WAVE 1 --------------------
-
+# WAVE 1
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if not title or not genre or not rating:
+        return None
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+def watch_movie(user_data, title):
+    for movies in user_data["watchlist"]:
+        if title == movies["title"]:
+            user_data["watched"].append(movies)
+            user_data["watchlist"].remove(movies)
+    return user_data
 
+# WAVE 2
+def get_watched_avg_rating(user_data):
+    ratings = 0.0
+    count_movies = 0
+    if not user_data["watched"]:
+        return ratings 
+    for movie_info in user_data["watched"]:
+        if isinstance(movie_info.get("rating"),float):
+            ratings += movie_info.get("rating")
+            count_movies += 1
+    return ratings/count_movies
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
+def get_most_watched_genre(user_data):
+    favorite_genre ={}

```

Comment:
[naming] I'd pick something like `genre_counts` for this. It holds multiple genres, not just a single favorite genre.